### DISCUSSION TITLE
Rule 3 again - thoughts and ideas

### DISCUSSION THREAD

**Discussion by ghost** [modrinth/.github#119] [Other discussions]
[Rule 3 (cheats and hacks)](https://modrinth.com/legal/rules) is flawed, at least in my opinion. It doesn't well define the threshold of what is/should be enforced clearly enough (which causes occasional debates within the community) and makes Modrinth more restrictive than the competitors (which leads to some modders sticking to CF when they would've uploaded to MR be the rule more lenient). It puts responsibility on mod authors to follow rules and on Modrinth - to enforce them, rather than on the player to follow rules set and enforced by the servers they play on.

A while ago in a conversation with triphora about this there was brought up an issue that makes this rule needed: the "nazi bar" problem. Essentially, if you let people from an unwanted demographic stay, even if they don't cause issues, you might attract more of this demographic and become known for housing it, like it or not. In Modrinth's case, we shouldn't let apparent cheats go public on the platform, as this might cause more of the same kind to start coming in and start making the platform an unwanted reputation for housing cheats.

I would love to see the team elaborate on other, if any, purposes of the rule.

This said, being presented with this issue, I won't advocate for removing the rule altogether, however it should still be reworked to set the bar more clearly and precisely. That bar would be just enough to match common sence of what a project with no major intended use other than to provide a game-changingly unfair advantage in multiplayer is, as that's what we're (probably) trying to avoid in the first place to avoid the potential reputational damage coming with hosting such mods.

So here's the *core* wording I came up with that I believe would work better for what's needed: *a cheat is a client-side modification designed to provide game-changingly advantageous information or abilities unavaivable to players without a comparable modification in a multiplayer context.*  
Admittedly, it's definitely not perfect, but it gets my point across.

Again, please discuss what issues the rule needs to adress, as I know I might not have the full picture here.

**Comment by Fnige** [comment]
Ima be honest I think the rule is defined well enough, could it be better? Sure but that goes for everything.

**Reply by Fnige** [reply]
Ok scratch that, the cheating rule in question only describes client side cheats. Given a certain conversation maybe it should also describe other kinds of cheats, like ones on the server

**Reply by AmyMialee** [reply]
There is no such thing as server side cheats, the rules of any given situation are defined by the server, so a server can't cheat as the standards would always match what the server allows.
Anything described as a server side cheat is just something that superficially resembles the features of a client side cheat, but isn't cheating as it abides by whatever rules the server owner has laid out.

It's impossible to police too, as many mod features give you abilities that on a client side mod would be cheating, but through the lens of a mod, like jetpacks for flying or spelunker potion for xray.

**Reply by AmyMialee** [reply]
And you can't really draw any strong line between what is and isn't, like you could make a server side mod to enable creative flight with a command, or by default, and it wouldn't have any of the visual dressing that a jetpack would.

Server Side Cheats don't exist.

**Reply by ghost** [reply]
The conversation in question was more about whether MR wanted to host morally shady plugins (more specifically, a plugin made to create fake players and rig the player count display)

**Reply by AmyMialee** [reply]
Yea but you could apply that to mods like Carpet too I'd assume.
It's very hard to draw very strong lines around this stuff.
